src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/amg/link.py

```diff
@@ -28,7 +28,23 @@ def link_grafana_instance(cmd, raw_parameters, azure_monitor_workspace_resource_
         )
         headers = ['User-Agent=azuremonitormetrics.link_grafana_instance']
         grafanaArmResponse = send_raw_request(cmd.cli_ctx, "GET", grafanaURI, body={}, headers=headers)
-        servicePrincipalId = grafanaArmResponse.json()["identity"]["principalId"]
+
+        # Check if 'identity' and 'type' exist in the response
+        identity_info = grafanaArmResponse.json().get("identity", {})
+        identity_type = identity_info.get("type", "").lower()
+
+        if identity_type == "systemassigned":
+            servicePrincipalId = identity_info.get("principalId")
+        elif identity_type == "userassigned":
```

Comment:
Do we need to convert the param into lowercase and then compare them?
```suggestion
        if identity_type.lower() == "systemassigned":
            servicePrincipalId = identity_info.get("principalId")
        elif identity_type.lower() == "userassigned":
```

Reply:
@yanzhudd  I'm converting the parameter to lowercase on line 34. That should suffice for this comparison, I believe?